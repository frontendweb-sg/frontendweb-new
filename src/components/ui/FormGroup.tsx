import classNames from "classnames";
import { Align } from "@/types";
import { forwardRef } from "react";

/**
 * Form group component
 * @param param0
 * @returns
 */
export interface FormGropProps
	extends React.HtmlHTMLAttributes<HTMLDivElement> {
	align?: Align;
	label?: string;
}
const FormGroup = forwardRef<HTMLDivElement, FormGropProps>(
	({ children, className, align, label, ...rest }, ref) => {
		const classes = classNames("form-group mb-3", "text-" + align, className);
		return (
			<div data-testid="form-group" className={classes} ref={ref} {...rest}>
				{label && <label className="form-label">{label}</label>}
				{children}
			</div>
		);
	}
);

export default FormGroup;
