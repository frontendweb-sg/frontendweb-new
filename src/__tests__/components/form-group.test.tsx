import FormGroup, { FormGropProps } from "@/components/ui/FormGroup";
import { render, screen } from "@/test-util";

const setup = (props: FormGropProps) =>
	render(<FormGroup {...props}>test</FormGroup>);

describe("FormGroup", () => {
	it("should render", () => {
		const { container } = setup({});
		expect(container).toMatchSnapshot();
	});

	it("should render with children", () => {
		setup({});
		expect(screen.getByText("test")).toBeInTheDocument();
	});

	it("should render children with label", () => {
		setup({ label: "Hello world" });
		expect(screen.getByText("Hello world")).toBeInTheDocument();
	});

	it("should render with align left", () => {
		setup({ align: "left" });
		expect(screen.getByTestId("form-group")).toHaveClass("text-left");
	});

	it("should render with custom className", () => {
		setup({ className: "test-class" });
		expect(screen.getByTestId("form-group")).toHaveClass("test-class");
	});
});
