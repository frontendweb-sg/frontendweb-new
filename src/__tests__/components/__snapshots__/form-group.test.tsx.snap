// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGroup should render 1`] = `
<div>
  <div
    class="form-group mb-3 text-undefined"
    data-testid="form-group"
  >
    test
  </div>
</div>
`;
